### PR TITLE
Developer Guide: Update Product Scope, User Stories, Non-Functional Requirements

### PR DESCRIPTION
**Pull request by shunyao643**
WIP

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -134,17 +134,26 @@ The target user is the logistics chief of an event support club. He is responsib
...
 --------------------------------------------------------------------------------------------------------------------
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ...                              | I want to ...                                                  | So that I can ...                                                    |
+|---------|---------------------------------------|----------------------------------------------------------------|----------------------------------------------------------------------|
+| v_.0    | new user                              | see usage instructions                                         | refer to them when I forget how to use the application               |
```
Comment (yuhaochua) [inline docs/DeveloperGuide.md]: I think for this, we can write as v2.0?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -134,17 +134,26 @@ The target user is the logistics chief of an event support club. He is responsib
...
+| v_.0    | new user                              | list all equipment                                             | see all current equipment in the application                         |
+| v_.0    | conscientious logs chief              | add equipment                                                  | newly purchased equipment can be kept up to date                     |
```
Comment (yuhaochua) [inline docs/DeveloperGuide.md]: I think for list and add equipment, we can write as v1.0?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -134,17 +134,26 @@ The target user is the logistics chief of an event support club. He is responsib
...
+| v_.0    | meticulous logs chief                 | update equipment                                               | correct any errors in previously recorded equipment                  |
+| v_.0    | conscientious logs chief              | delete equipment                                               | remove decommissioned equipment from the application                 |
+| v_.0    | forgetful logs chief                  | search for equipment                                           | be able to find equipment by its attributes                          |
```
Comment (yuhaochua) [inline docs/DeveloperGuide.md]: I think for this remaining 3 user stories, we can fill in as v2.1?

**Review by yuhaochua — COMMENTED** [review]
I just have some suggestions for the version remarks for user stories, otherwise the changes look good so far!

**Review by yuhaochua — APPROVED** [review]
LGTM, thanks for refining all the sections of developer guide and filling in the remaining portions!

**Review by Bryan-BC — APPROVED** [review]
LGTM. Thanks for updating the DG

**Pull request merged**